corehq/apps/users/static/users/js/filtered_download.js

```diff
@@ -37,16 +39,23 @@ hqDefine('users/js/filtered_download', [
             });
         });
 
+        self.location_change = function () {
+            self.location_id($('#id_location_id').val());
```

Comment:
Generally UIs should be controlled by knockout or jQuery, not a mix of the two. Crispy forms has knockout support: you can add a `self.location_id = ko.observable()` above, then add a `data_bind="value: location_id"` argument to the location id field in `UserFilterForm`, and then this function won't be needed.

Reply:
So, interestingly, I did do that initially. In fact, the `data_bind` code is still there, but when I inspect the value that's supposed to be bound I get an `undefined`, thus the subscribe does not work so the subscription to `countUsers` is never fired when I select another location. I spent quite some time trying to figure out why and ended up doing this jQuery workaround.

Maybe I'm missing something obvious...?

Reply:
Interesting indeed...I was curious why that hadn't worked, so I checked out your branch, and found a rather non-obvious issue. Turns out that the HTML template for the `LocationSelectWidget` doesn't handle arbitrary knockout bindings. Take a look at [this commit](https://github.com/dimagi/commcare-hq/commit/65a0248228e5de98c00174677c5a16a1876d9f5d) (from [this branch](https://github.com/dimagi/commcare-hq/compare/cs/SC-1567-bulk-export-filters-debug), which just branches off this PR) which adds that support.

One note on that commit - I added a new `data_bind` kwarg to the `LocationSelectWidget` constructor instead of using the widget's built-in `attrs` property because `attrs` typically uses the same names as the real HTML attributes, so you'd pass `attrs={"data-bind": "value: location_id"}`...but then you can't reference `attrs.data-bind` in the HTML template because of the hyphen.

Reply:
@orangejenny Sorry for only responding now. 

Ah, I see. Retrospectively very obvious. I updated the branch with the data bind code and removed the jQuery. Looks like it works well.